### PR TITLE
dev-python/*: bumps to python 3.10

### PR DESCRIPTION
**Pull request by arthurzam**
@mgorny Sorry for the messy PR, but it was a collection of various tries and bumps to python 3.10. Some of them were forgotten and I found them in git stash...

I'm not that sure when can we bump to `EAPI=8`? I know that packages that have stable version shouldn't be bumped to `EAPI=8` for a year, so only for one package I bumped the EAPI.

1. `dev-python/eyeD3` passes tests and trivial bump
2. `dev-python/pytest-tempdir` passes tests, did small cleanup
3. `dev-python/xxhash` keyworded for x86 - passed tests for me using virtual x86 system. Needed for next package...
4. `dev-python/kafka-python` - big work to improve various things - Needs extra attention when reviewing!
5. `dev-python/exam` trivial bump and passes tests
6. `dev-python/pyhcl`  trivial bump and passes tests
7. `dev-util/diff-cover` wanted to bump, but failed, but lets improve metadata :)
8. `dev-python/isodate` wanted to bump, but failed, but lets improve metadata :)
9. `dev-python/fleep` bump python and EAPI. Tested by running examples on multiple files on my system.
10. `dev-python/emoji` trivial bump and passes tests

### CODE REVIEW

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: dev-python/fleep/fleep-9999.ebuild [hunk 1]
```diff
@@ -20,4 +20,3 @@ fi
 
 LICENSE="MIT"
```
[the remark this answers is not in the document]
Comment (arthurzam) [reply dev-python/fleep/fleep-9999.ebuild:21]: Fixed bash test.

About testsuite, I need to change to github for source files. The test suite itself is very basic and nearly doesn't check a lot (only one file). I think adding it isn't worth it (meaning we need an `"${P}.gh.tar.gz"` src file).

But if you still prefer to add tests, I will do so :)

**Comment by mgorny** [comment]
> @mgorny Sorry for the messy PR, but it was a collection of various tries and bumps to python 3.10. Some of them were forgotten and I found them in git stash...
> 
> I'm not that sure when can we bump to `EAPI=8`? I know that packages that have stable version shouldn't be bumped to `EAPI=8` for a year, so only for one package I bumped the EAPI.

No such rule. I generally bump everything to `EAPI=8` when I touch it (without revbump, it's a "boring" EAPI), unless it's really complex.

**Comment by arthurzam** [comment]
> > @mgorny Sorry for the messy PR, but it was a collection of various tries and bumps to python 3.10. Some of them were forgotten and I found them in git stash...
> > I'm not that sure when can we bump to `EAPI=8`? I know that packages that have stable version shouldn't be bumped to `EAPI=8` for a year, so only for one package I bumped the EAPI.
> 
> No such rule. I generally bump everything to `EAPI=8` when I touch it (without revbump, it's a "boring" EAPI), unless it's really complex.

OK, will do the same. I was sure such rule was mentioned in gentoo-dev at some point - I guess I misread.

File: dev-python/exam/exam-0.10.6.ebuild [hunk 2]
```diff
@@ -1,9 +1,9 @@
-# Copyright 1999-2020 Gentoo Authors
```
Comment (mgorny) [inline dev-python/exam/exam-0.10.6.ebuild:1]: For the future: if you see nose, check if the package in question actually uses it (i.e. `import`s anything from nose). If not, try to swap it for unittest or pytest. (I'll do it here)

File: dev-python/emoji/emoji-1.2.1.ebuild [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7,8,9} )
```
Comment (mgorny) [inline dev-python/emoji/emoji-1.2.1.ebuild:6]: This is already handled in 1.4.1 bump, so I'll skip this commit.

File: dev-python/fleep/fleep-9999.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgorny) [reply dev-python/fleep/fleep-9999.ebuild:21]: Yes, please. I'll skip this commit then too. Feel free to either rebase with this one package or move it to another PR.

File: dev-python/exam/exam-0.10.6.ebuild (hunk 2 above)
Comment (arthurzam) [reply dev-python/exam/exam-0.10.6.ebuild]: Thanks for the note - will check in the future.

**Comment by arthurzam** [comment]
I have added the testsuite. It is **very basic**.

I bumped the revision, and added a temporary note to remove int he future the `.gh` suffix. Of course this note doesn't exist in the live version (I hope the future person will copy the live version to the new version).

File: dev-python/fleep/fleep-1.0.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+PYTHON_COMPAT=( python3_{8..10} )
+
+inherit distutils-r1
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://github.com/floyernick/fleep-py.git"
+	inherit git-r3
+else
+	# upstream aren't tagging the releases, just as commit title
```
Comment (mgorny) [inline dev-python/fleep/fleep-1.0.1-r1.ebuild:14]: File a bug!

Comment (arthurzam) [reply dev-python/fleep/fleep-1.0.1-r1.ebuild:14]: Opened here https://github.com/floyernick/fleep-py/issues/17

**Pull request closed without merging**